### PR TITLE
ci: move E2E runner from github to AWS

### LINKED ISSUE #250: Convert E2E job to use AWS runners
See https://github.com/instructlab/instructlab/issues/2036 for reference

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Resolves https://github.com/instructlab/sdg/issues/250

### CODE REVIEW

**Comment by alimaredia** [comment]
@nathan-weinberg can you send a run triggered by hand that shows that these new runners work?

**Comment by nathan-weinberg** [comment]
> @nathan-weinberg can you send a run triggered by hand that shows that these new runners work?

I have no way of doing that, you can looking at the Training implementation for reference

File: .github/workflows/e2e-nvidia-t4-x1.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          python3.11 -m pip install .[cuda]
+        
+      - name: Update instructlab-sdg library
+        working-directory: ./sdg
+        run: |
+          . ./instructlab/venv/bin/activate
+          pip install .
+          pip install .[cuda]        
+
+      - name: Run e2e test
```
Comment (danmcp) [inline .github/workflows/e2e-nvidia-t4-x1.yml:131]: Needs `working-directory: ./instructlab`

File: .github/workflows/e2e-nvidia-t4-x1.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+          # https://github.com/instructlab/instructlab/issues/1821
+          # install with Torch and build dependencies installed
+          python3.11 -m pip install packaging wheel setuptools-scm
+          python3.11 -m pip install .[cuda]
+        
+      - name: Update instructlab-sdg library
+        working-directory: ./sdg
+        run: |
+          . ./instructlab/venv/bin/activate
```
Comment (danmcp) [inline .github/workflows/e2e-nvidia-t4-x1.yml:127]: Should be: `. ../instructlab/venv/bin/activate`

**Review by danmcp — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-nvidia-t4-x1.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+          pip install .
+          pip install .[cuda]        
+
+      - name: Run e2e test
+        run: |
+          . venv/bin/activate
+          ./instructlab/scripts/basic-workflow-tests.sh -m
```
Comment (danmcp) [inline .github/workflows/e2e-nvidia-t4-x1.yml:134]: And this one `./scripts/basic-workflow-tests.sh -m`

**Review by danmcp — APPROVED** [review]

**Review by khaledsulayman — APPROVED** [review]

**Pull request merged**